src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -39,6 +40,12 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 !string.IsNullOrEmpty(bindingContext.ModelName);
 
             var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
+            if (newBindingContext == null)
+            {
+                // Unable to find a value provider for this binding source. Binding will fail.
+                return null;
+            }
```

Comment:
This is a really roundabout way of checking this - will we really fail if there's no VP? a model can have all greedy properties.